staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -257,8 +263,8 @@ func finishRequest(timeout time.Duration, fn resultFunc) (result runtime.Object,
 		return nil, err
 	case p := <-panicCh:
 		panic(p)
-	case <-time.After(timeout):
-		return nil, errors.NewTimeoutError(fmt.Sprintf("request did not complete within requested timeout %s", timeout), 0)
+	case <-ctx.Done():
+		return nil, errors.NewTimeoutError(fmt.Sprintf("request did not complete within requested timeout %s", ctx.Err()), 0)
```

Comment:
split the cases of `Err()` for timeout or cancelled please.

Reply:
@deads2k am i missing a use case here? in either case, the request has timed out whether it's a context with a deadline or with a cancel.

Reply:
> @deads2k am i missing a use case here? in either case, the request has timed out whether it's a context with a deadline or with a cancel.

As a dev debugging something for a cluster-admin, I want to know exactly what happened the information exists right here and is easy to expose.